src/story.ts

```diff
@@ -0,0 +1,31 @@
+export type StoryId = string;
+export type StoryKind = string;
+export type StoryName = string;
+
+type StoryReturnType<T> = T extends (storyFn: infer StoryFn, context: any) => any
```

Comment:
Can you explain this a bit? I'm not familiar with `infer` and the whole thing looks pretty tricky. (Not wrong, just want to understand better!)

Reply:
Yeah, sure. Think about this like you extract some internal type of prop/arg from another complex type. So if `T` generic match with type declared after `extends` keyword then you can use type that you inferred from `T`. In first true condition also used [conditional types](https://mariusschulz.com/blog/conditional-types-in-typescript).

Here is explanation for `infer` keyword https://dev.to/aexol/typescript-tutorial-infer-keyword-2cn